Export fails in CLI, Never finishes in GUI

Error when exporting any user.
When trying to export on the GUI, the export never finishes, even after days. When using the CLI, it fails on trying to export files.

Below is the output with max verbosity when trying via CLI:

```
www-data@895183ca82fc:~/html$ php occ user:export j.hildebrandt data/ -vvv
Exporting user information in user.json…
Exporting settings in settings.json…
Exporting versions in versions.json…
Exporting calendars into dav/calendars/…
Exporting contacts into dav/address_books/…
Exporting trashbin into files_trashbin…
Exporting trashbin files…
Exporting account information in settings/account.json…
Exporting profile config in settings/config.json…
Exporting files…
Exporting file versions…

An unhandled exception has been thrown:
TypeError: Symfony\Component\Console\Formatter\OutputFormatter::escapeTrailingBackslash(): Argument #1 ($text) must be of type string, array given, called in /var/www/html/3rdparty/symfony/console/Formatter/OutputFormatter.php on line 47 and defined in /var/www/html/3rdparty/symfony/console/Formatter/OutputFormatter.php:55
Stack trace:
#0 /var/www/html/3rdparty/symfony/console/Formatter/OutputFormatter.php(47): Symfony\Component\Console\Formatter\OutputFormatter::escapeTrailingBackslash(Array)
#1 /var/www/html/3rdparty/symfony/console/Style/SymfonyStyle.php(471): Symfony\Component\Console\Formatter\OutputFormatter::escape(Array)
#2 /var/www/html/3rdparty/symfony/console/Style/SymfonyStyle.php(72): Symfony\Component\Console\Style\SymfonyStyle->createBlock(Array, '[ERROR] ', 'fg=white;bg=red', ' ', true, true)
#3 /var/www/html/3rdparty/symfony/console/Style/SymfonyStyle.php(152): Symfony\Component\Console\Style\SymfonyStyle->block(Array, 'ERROR', 'fg=white;bg=red', ' ', true)
#4 /var/www/html/apps/user_migration/lib/Command/Export.php(215): Symfony\Component\Console\Style\SymfonyStyle->error(Array)
#5 /var/www/html/3rdparty/symfony/console/Command/Command.php(255): OCA\UserMigration\Command\Export->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#6 /var/www/html/core/Command/Base.php(177): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#7 /var/www/html/3rdparty/symfony/console/Application.php(1009): OC\Core\Command\Base->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#8 /var/www/html/3rdparty/symfony/console/Application.php(273): Symfony\Component\Console\Application->doRunCommand(Object(OCA\UserMigration\Command\Export), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#9 /var/www/html/3rdparty/symfony/console/Application.php(149): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#10 /var/www/html/lib/private/Console/Application.php(213): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 /var/www/html/console.php(100): OC\Console\Application->run()
#12 /var/www/html/occ(11): require_once('/var/www/html/c...')
#13 {main}www-data@895183ca82fc:~/html$ 
```


And below the respective Log entries from nextcloud.log
```
www-data@895183ca82fc:~/html$ tail data/nextcloud.log | grep migra
{"reqId":"VCLDSeAptVTEy7OmpZ7Z","level":3,"time":"2023-02-25T13:47:31+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"Array to string conversion at /var/www/html/3rdparty/symfony/console/Formatter/OutputFormatter.php#45","userAgent":"--","version":"25.0.3.2","exception":{"Exception":"Error","Message":"Array to string conversion at /var/www/html/3rdparty/symfony/console/Formatter/OutputFormatter.php#45","Code":0,"Trace":[{"function":"onError","class":"OC\\Log\\ErrorHandler","type":"::","args":[2,"Array to string conversion","/var/www/html/3rdparty/symfony/console/Formatter/OutputFormatter.php",45]},{"file":"/var/www/html/3rdparty/symfony/console/Formatter/OutputFormatter.php","line":45,"function":"preg_replace","args":["/([^\\\\]?)</","$1\\<",["/var/www/html/apps/user_migration/lib/Service/UserMigrationService.php",216,"export","OCA\\UserMigration\\Migrator\\FilesMigrator","->","And 1 more entries, set log level to debug to see all entries"]]},{"file":"/var/www/html/3rdparty/symfony/console/Style/SymfonyStyle.php","line":471,"function":"escape","class":"Symfony\\Component\\Console\\Formatter\\OutputFormatter","type":"::","args":[["/var/www/html/apps/user_migration/lib/Service/UserMigrationService.php",216,"export","OCA\\UserMigration\\Migrator\\FilesMigrator","->","And 1 more entries, set log level to debug to see all entries"]]},{"file":"/var/www/html/3rdparty/symfony/console/Style/SymfonyStyle.php","line":72,"function":"createBlock","class":"Symfony\\Component\\Console\\Style\\SymfonyStyle","type":"->","args":[[["/var/www/html/apps/user_migration/lib/Service/UserMigrationService.php",216,"export","OCA\\UserMigration\\Migrator\\FilesMigrator","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/apps/user_migration/lib/Command/Export.php",206,"export","OCA\\UserMigration\\Service\\UserMigrationService","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/3rdparty/symfony/console/Command/Command.php",255,"execute","OCA\\UserMigration\\Command\\Export","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/core/Command/Base.php",177,"run","Symfony\\Component\\Console\\Command\\Command","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/3rdparty/symfony/console/Application.php",1009,"run","OC\\Core\\Command\\Base","->","And 1 more entries, set log level to debug to see all entries"],"And 5 more entries, set log level to debug to see all entries"],"[ERROR] ","fg=white;bg=red"," ",true,true]},{"file":"/var/www/html/3rdparty/symfony/console/Style/SymfonyStyle.php","line":152,"function":"block","class":"Symfony\\Component\\Console\\Style\\SymfonyStyle","type":"->","args":[[["/var/www/html/apps/user_migration/lib/Service/UserMigrationService.php",216,"export","OCA\\UserMigration\\Migrator\\FilesMigrator","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/apps/user_migration/lib/Command/Export.php",206,"export","OCA\\UserMigration\\Service\\UserMigrationService","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/3rdparty/symfony/console/Command/Command.php",255,"execute","OCA\\UserMigration\\Command\\Export","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/core/Command/Base.php",177,"run","Symfony\\Component\\Console\\Command\\Command","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/3rdparty/symfony/console/Application.php",1009,"run","OC\\Core\\Command\\Base","->","And 1 more entries, set log level to debug to see all entries"],"And 5 more entries, set log level to debug to see all entries"],"ERROR","fg=white;bg=red"," ",true]},{"file":"/var/www/html/apps/user_migration/lib/Command/Export.php","line":215,"function":"error","class":"Symfony\\Component\\Console\\Style\\SymfonyStyle","type":"->","args":[[["/var/www/html/apps/user_migration/lib/Service/UserMigrationService.php",216,"export","OCA\\UserMigration\\Migrator\\FilesMigrator","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/apps/user_migration/lib/Command/Export.php",206,"export","OCA\\UserMigration\\Service\\UserMigrationService","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/3rdparty/symfony/console/Command/Command.php",255,"execute","OCA\\UserMigration\\Command\\Export","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/core/Command/Base.php",177,"run","Symfony\\Component\\Console\\Command\\Command","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/3rdparty/symfony/console/Application.php",1009,"run","OC\\Core\\Command\\Base","->","And 1 more entries, set log level to debug to see all entries"],"And 5 more entries, set log level to debug to see all entries"]]},{"file":"/var/www/html/3rdparty/symfony/console/Command/Command.php","line":255,"function":"execute","class":"OCA\\UserMigration\\Command\\Export","type":"->","args":[{"__class__":"Symfony\\Component\\Console\\Input\\ArgvInput"},{"__class__":"Symfony\\Component\\Console\\Output\\ConsoleOutput"}]},{"file":"/var/www/html/core/Command/Base.php","line":177,"function":"run","class":"Symfony\\Component\\Console\\Command\\Command","type":"->","args":[{"__class__":"Symfony\\Component\\Console\\Input\\ArgvInput"},{"__class__":"Symfony\\Component\\Console\\Output\\ConsoleOutput"}]},{"file":"/var/www/html/3rdparty/symfony/console/Application.php","line":1009,"function":"run","class":"OC\\Core\\Command\\Base","type":"->","args":[{"__class__":"Symfony\\Component\\Console\\Input\\ArgvInput"},{"__class__":"Symfony\\Component\\Console\\Output\\ConsoleOutput"}]},{"file":"/var/www/html/3rdparty/symfony/console/Application.php","line":273,"function":"doRunCommand","class":"Symfony\\Component\\Console\\Application","type":"->","args":[{"__class__":"OCA\\UserMigration\\Command\\Export"},{"__class__":"Symfony\\Component\\Console\\Input\\ArgvInput"},{"__class__":"Symfony\\Component\\Console\\Output\\ConsoleOutput"}]},{"file":"/var/www/html/3rdparty/symfony/console/Application.php","line":149,"function":"doRun","class":"Symfony\\Component\\Console\\Application","type":"->","args":[{"__class__":"Symfony\\Component\\Console\\Input\\ArgvInput"},{"__class__":"Symfony\\Component\\Console\\Output\\ConsoleOutput"}]},{"file":"/var/www/html/lib/private/Console/Application.php","line":213,"function":"run","class":"Symfony\\Component\\Console\\Application","type":"->","args":[{"__class__":"Symfony\\Component\\Console\\Input\\ArgvInput"},{"__class__":"Symfony\\Component\\Console\\Output\\ConsoleOutput"}]},{"file":"/var/www/html/console.php","line":100,"function":"run","class":"OC\\Console\\Application","type":"->","args":[]},{"file":"/var/www/html/occ","line":11,"args":["/var/www/html/console.php"],"function":"require_once"}],"File":"/var/www/html/lib/private/Log/ErrorHandler.php","Line":92,"CustomMessage":"--"}}
{"reqId":"kYpfuzPNLn0LNkwT8Vgj","level":3,"time":"2023-02-25T13:54:58+00:00","remoteAddr":"","user":"--","app":"PHP","method":"","url":"--","message":"Array to string conversion at /var/www/html/3rdparty/symfony/console/Formatter/OutputFormatter.php#45","userAgent":"--","version":"25.0.3.2","exception":{"Exception":"Error","Message":"Array to string conversion at /var/www/html/3rdparty/symfony/console/Formatter/OutputFormatter.php#45","Code":0,"Trace":[{"function":"onError","class":"OC\\Log\\ErrorHandler","type":"::","args":[2,"Array to string conversion","/var/www/html/3rdparty/symfony/console/Formatter/OutputFormatter.php",45]},{"file":"/var/www/html/3rdparty/symfony/console/Formatter/OutputFormatter.php","line":45,"function":"preg_replace","args":["/([^\\\\]?)</","$1\\<",["/var/www/html/apps/user_migration/lib/Service/UserMigrationService.php",216,"export","OCA\\UserMigration\\Migrator\\FilesMigrator","->","And 1 more entries, set log level to debug to see all entries"]]},{"file":"/var/www/html/3rdparty/symfony/console/Style/SymfonyStyle.php","line":471,"function":"escape","class":"Symfony\\Component\\Console\\Formatter\\OutputFormatter","type":"::","args":[["/var/www/html/apps/user_migration/lib/Service/UserMigrationService.php",216,"export","OCA\\UserMigration\\Migrator\\FilesMigrator","->","And 1 more entries, set log level to debug to see all entries"]]},{"file":"/var/www/html/3rdparty/symfony/console/Style/SymfonyStyle.php","line":72,"function":"createBlock","class":"Symfony\\Component\\Console\\Style\\SymfonyStyle","type":"->","args":[[["/var/www/html/apps/user_migration/lib/Service/UserMigrationService.php",216,"export","OCA\\UserMigration\\Migrator\\FilesMigrator","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/apps/user_migration/lib/Command/Export.php",206,"export","OCA\\UserMigration\\Service\\UserMigrationService","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/3rdparty/symfony/console/Command/Command.php",255,"execute","OCA\\UserMigration\\Command\\Export","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/core/Command/Base.php",177,"run","Symfony\\Component\\Console\\Command\\Command","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/3rdparty/symfony/console/Application.php",1009,"run","OC\\Core\\Command\\Base","->","And 1 more entries, set log level to debug to see all entries"],"And 5 more entries, set log level to debug to see all entries"],"[ERROR] ","fg=white;bg=red"," ",true,true]},{"file":"/var/www/html/3rdparty/symfony/console/Style/SymfonyStyle.php","line":152,"function":"block","class":"Symfony\\Component\\Console\\Style\\SymfonyStyle","type":"->","args":[[["/var/www/html/apps/user_migration/lib/Service/UserMigrationService.php",216,"export","OCA\\UserMigration\\Migrator\\FilesMigrator","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/apps/user_migration/lib/Command/Export.php",206,"export","OCA\\UserMigration\\Service\\UserMigrationService","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/3rdparty/symfony/console/Command/Command.php",255,"execute","OCA\\UserMigration\\Command\\Export","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/core/Command/Base.php",177,"run","Symfony\\Component\\Console\\Command\\Command","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/3rdparty/symfony/console/Application.php",1009,"run","OC\\Core\\Command\\Base","->","And 1 more entries, set log level to debug to see all entries"],"And 5 more entries, set log level to debug to see all entries"],"ERROR","fg=white;bg=red"," ",true]},{"file":"/var/www/html/apps/user_migration/lib/Command/Export.php","line":215,"function":"error","class":"Symfony\\Component\\Console\\Style\\SymfonyStyle","type":"->","args":[[["/var/www/html/apps/user_migration/lib/Service/UserMigrationService.php",216,"export","OCA\\UserMigration\\Migrator\\FilesMigrator","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/apps/user_migration/lib/Command/Export.php",206,"export","OCA\\UserMigration\\Service\\UserMigrationService","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/3rdparty/symfony/console/Command/Command.php",255,"execute","OCA\\UserMigration\\Command\\Export","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/core/Command/Base.php",177,"run","Symfony\\Component\\Console\\Command\\Command","->","And 1 more entries, set log level to debug to see all entries"],["/var/www/html/3rdparty/symfony/console/Application.php",1009,"run","OC\\Core\\Command\\Base","->","And 1 more entries, set log level to debug to see all entries"],"And 5 more entries, set log level to debug to see all entries"]]},{"file":"/var/www/html/3rdparty/symfony/console/Command/Command.php","line":255,"function":"execute","class":"OCA\\UserMigration\\Command\\Export","type":"->","args":[{"__class__":"Symfony\\Component\\Console\\Input\\ArgvInput"},{"__class__":"Symfony\\Component\\Console\\Output\\ConsoleOutput"}]},{"file":"/var/www/html/core/Command/Base.php","line":177,"function":"run","class":"Symfony\\Component\\Console\\Command\\Command","type":"->","args":[{"__class__":"Symfony\\Component\\Console\\Input\\ArgvInput"},{"__class__":"Symfony\\Component\\Console\\Output\\ConsoleOutput"}]},{"file":"/var/www/html/3rdparty/symfony/console/Application.php","line":1009,"function":"run","class":"OC\\Core\\Command\\Base","type":"->","args":[{"__class__":"Symfony\\Component\\Console\\Input\\ArgvInput"},{"__class__":"Symfony\\Component\\Console\\Output\\ConsoleOutput"}]},{"file":"/var/www/html/3rdparty/symfony/console/Application.php","line":273,"function":"doRunCommand","class":"Symfony\\Component\\Console\\Application","type":"->","args":[{"__class__":"OCA\\UserMigration\\Command\\Export"},{"__class__":"Symfony\\Component\\Console\\Input\\ArgvInput"},{"__class__":"Symfony\\Component\\Console\\Output\\ConsoleOutput"}]},{"file":"/var/www/html/3rdparty/symfony/console/Application.php","line":149,"function":"doRun","class":"Symfony\\Component\\Console\\Application","type":"->","args":[{"__class__":"Symfony\\Component\\Console\\Input\\ArgvInput"},{"__class__":"Symfony\\Component\\Console\\Output\\ConsoleOutput"}]},{"file":"/var/www/html/lib/private/Console/Application.php","line":213,"function":"run","class":"Symfony\\Component\\Console\\Application","type":"->","args":[{"__class__":"Symfony\\Component\\Console\\Input\\ArgvInput"},{"__class__":"Symfony\\Component\\Console\\Output\\ConsoleOutput"}]},{"file":"/var/www/html/console.php","line":100,"function":"run","class":"OC\\Console\\Application","type":"->","args":[]},{"file":"/var/www/html/occ","line":11,"args":["/var/www/html/console.php"],"function":"require_once"}],"File":"/var/www/html/lib/private/Log/ErrorHandler.php","Line":92,"CustomMessage":"--"}}

```

I'm not sure how to proceed, the error message is greek to me.

--- a/lib/private/SystemTag/SystemTagObjectMapper.php
+++ b/lib/private/SystemTag/SystemTagObjectMapper.php
@@ -77,23 +77,25 @@ class SystemTagObjectMapper implements ISystemTagObjectMapper {
 			->from(self::RELATION_TABLE)
 			->where($query->expr()->in('objectid', $query->createParameter('objectids')))
 			->andWhere($query->expr()->eq('objecttype', $query->createParameter('objecttype')))
-			->setParameter('objectids', $objIds, IQueryBuilder::PARAM_STR_ARRAY)
 			->setParameter('objecttype', $objectType)
 			->addOrderBy('objectid', 'ASC')
 			->addOrderBy('systemtagid', 'ASC');
-
+		$chunks = array_chunk($objIds, 900, false);
 		$mapping = [];
 		foreach ($objIds as $objId) {
 			$mapping[$objId] = [];
 		}
+		foreach ($chunks as $chunk) {
+			$query->setParameter('objectids', $chunk, IQueryBuilder::PARAM_STR_ARRAY);
+			$result = $query->executeQuery();
+			while ($row = $result->fetch()) {
+				$objectId = $row['objectid'];
+				$mapping[$objectId][] = $row['systemtagid'];
+			}
 
-		$result = $query->execute();
-		while ($row = $result->fetch()) {
-			$objectId = $row['objectid'];
-			$mapping[$objectId][] = $row['systemtagid'];
+			$result->closeCursor();
 		}
 
-		$result->closeCursor();
 
 		return $mapping;
 	}

--- a/tests/lib/SystemTag/SystemTagObjectMapperTest.php
+++ b/tests/lib/SystemTag/SystemTagObjectMapperTest.php
@@ -141,6 +141,17 @@ class SystemTagObjectMapperTest extends TestCase {
 		$this->assertEquals([], $tagIdMapping);
 	}
 
+	public function testGetTagIdsForALotOfObjects() {
+		$ids = range(1, 10500);
+		$tagIdMapping = $this->tagMapper->getTagIdsForObjects(
+			$ids,
+			'testtype'
+		);
+
+		$this->assertEquals(10500, count($tagIdMapping));
+		$this->assertEquals([$this->tag1->getId(), $this->tag2->getId()], $tagIdMapping[1]);
+	}
+
 	public function testGetObjectsForTags() {
 		$objectIds = $this->tagMapper->getObjectIdsForTags(
 			[$this->tag1->getId(), $this->tag2->getId(), $this->tag3->getId()],
@@ -165,7 +176,7 @@ class SystemTagObjectMapperTest extends TestCase {
 		], $objectIds);
 	}
 
-	
+
 	public function testGetObjectsForTagsLimitWithMultipleTags() {
 		$this->expectException(\InvalidArgumentException::class);
 
@@ -189,7 +200,7 @@ class SystemTagObjectMapperTest extends TestCase {
 		], $objectIds);
 	}
 
-	
+
 	public function testGetObjectsForNonExistingTag() {
 		$this->expectException(\OCP\SystemTag\TagNotFoundException::class);
 
@@ -227,7 +238,7 @@ class SystemTagObjectMapperTest extends TestCase {
 		$this->assertTrue(true, 'No error when reassigning/unassigning');
 	}
 
-	
+
 	public function testAssignNonExistingTags() {
 		$this->expectException(\OCP\SystemTag\TagNotFoundException::class);
 
@@ -254,7 +265,7 @@ class SystemTagObjectMapperTest extends TestCase {
 		], $tagIdMapping, 'None of the tags got assigned');
 	}
 
-	
+
 	public function testUnassignNonExistingTags() {
 		$this->expectException(\OCP\SystemTag\TagNotFoundException::class);
 
@@ -385,7 +396,7 @@ class SystemTagObjectMapperTest extends TestCase {
 		);
 	}
 
-	
+
 	public function testHaveTagNonExisting() {
 		$this->expectException(\OCP\SystemTag\TagNotFoundException::class);
 

